File: whatsapp-clone-ui/src/app/pages/main/notification.ts
export interface Notification {
    chatId?: string;
    content?: string;
    senderId?: string;
    receiverId?: string;
    messageType?: 'TEXT' | 'IMAGE' | 'VIDEO' | 'AUDIO';
    type?: 'SEEN' | 'MESSAGE' | 'IMAGE' | 'VIDEO' | 'AUDIO';
    chatName?: string;
    media?: Array<string>;
}

File: whatsapp-clone-api/src/main/java/com/koula/whatsappclone/notification/Notification.java
package com.koula.whatsappclone.notification;

import com.koula.whatsappclone.message.MessageType;
import lombok.*;

@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class Notification {
    private String chatId;
    private String content;
    private String senderId;
    private String receiverId;
    private String chatName;
    private MessageType messageType;
    private NotificationType type;
    private byte[] media;
}
